cpp/src/arrow/compute/kernels/CMakeLists.txt

```diff
@@ -137,4 +137,5 @@ add_arrow_compute_test(aggregate_test
 # ----------------------------------------------------------------------
 # Utilities
 
-add_arrow_compute_test(kernel_utility_test SOURCES codegen_internal_test.cc)
+add_arrow_compute_test(kernel_utility_test SOURCES codegen_internal_test.cc
```

Comment:
It seems weird to add row encoder test as part of the kernel utility suite. If you do a search, you can find that row encoder is not referenced in `compute/kernels` directory at all. (The only call sites are in `compute/row/grouper.cc` and `acero/hash_join*.cc`.)

Maybe we should consider moving it out to `compute` directory. And we can make this test part of `arrow-compute-internals-test`.